pkg/strategy/record/strategy.go

```diff
@@ -0,0 +1,238 @@
+package record
+
+import (
+	"context"
+	"fmt"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/pkg/errors"
+	"sync"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/service"
+	"github.com/c9s/bbgo/pkg/types"
+	"github.com/c9s/bbgo/pkg/util"
+	log "github.com/sirupsen/logrus"
+	"github.com/slack-go/slack"
+)
+
+const ID = "record"
+
+const stateKey = "record"
+
+func init() {
+	bbgo.RegisterStrategy(ID, &Strategy{})
+}
+
+type State struct {
+	Since int64 `json:"since"`
+}
+
+func (s *State) PlainText() string {
+	return util.Render(`{{ .Asset }} transfer stats:
+daily number of transfers: {{ .DailyNumberOfTransfers }}
+daily amount of transfers {{ .DailyAmountOfTransfers.Float64 }}`, s)
+}
+
+func (s *State) SlackAttachment() slack.Attachment {
+	return slack.Attachment{
+		// Pretext:       "",
+		// Text:  text,
+		Fields: []slack.AttachmentField{},
+		Footer: util.Render("Since {{ . }}", time.Unix(s.Since, 0).Format(time.RFC822)),
+	}
+}
+
+func (s *State) Reset() {
+	var beginningOfTheDay = util.BeginningOfTheDay(time.Now().Local())
+	*s = State{
+		Since: beginningOfTheDay.Unix(),
+	}
+}
+
+type Strategy struct {
+	Notifiability *bbgo.Notifiability
+	*bbgo.Graceful
+	*bbgo.Persistence
+	AccountService *service.AccountService
+
+	Interval      types.Duration `json:"interval"`
+	ReportOnStart bool           `json:"reportOnStart"`
+	IgnoreDusts   bool           `json:"ignoreDusts"`
+	state         *State
+}
+
+func (s *Strategy) ID() string {
+	return ID
+}
+
+func (s *Strategy) CrossSubscribe(sessions map[string]*bbgo.ExchangeSession) {}
+
+func (s *Strategy) recordNetAssetValue(ctx context.Context, sessions map[string]*bbgo.ExchangeSession) {
+	totalAssets := types.AssetMap{}
+	totalBalances := types.BalanceMap{}
+	lastPrices := map[string]float64{}
+
+	notifyText := ""
+
+	time := time.Now().Local()
+
+	for _, session := range sessions {
+
+		exBalances, err := session.Exchange.QueryAccountBalances(ctx)
+		if err != nil {
+			message := fmt.Sprintf("update balance fail: %s %s ", session.ExchangeName, session.SubAccount)
+			s.Notifiability.Notify(message)
+			log.WithError(err).Error(message)
+			continue
+		}
+		session.Account.UpdateBalances(exBalances)
+
+		balances := session.Account.Balances()
+		var sessionBalances = types.BalanceMap{}
+
+		if err := session.UpdatePrices(ctx); err != nil {
+			log.WithError(err).Error("price update failed")
+			return
+		}
+
+		for _, b := range balances {
+			if tb, ok := sessionBalances[b.Currency]; ok {
+				tb.Available += b.Available
+				tb.Locked += b.Locked
+				sessionBalances[b.Currency] = tb
+			} else {
+				sessionBalances[b.Currency] = b
+			}
+		}
+
+		for _, b := range balances {
+			if tb, ok := totalBalances[b.Currency]; ok {
+				tb.Available += b.Available
+				tb.Locked += b.Locked
+				totalBalances[b.Currency] = tb
+			} else {
+				totalBalances[b.Currency] = b
+			}
+		}
+
+		prices := session.LastPrices()
```

Comment:
`session.LastPrices` now returns `map[string]fixedpoint.Value`.  
try to define `lastPrices` of the same type.